.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,64 @@
+name: Build and Test
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types:
+      - opened
+      - synchronize
+jobs:
+  build-and-test:
+    strategy:
+      matrix:
+        ghc: ["8.0.2", "8.2.2", "8.4.4", "8.6.5", "8.8.3", "8.10.1"]
+        os: [ubuntu-latest]
+        include:
+          - os: macos-latest
+            ghc: "8.10.1"
+          - os: ubuntu-16.04
+            ghc: "7.10.3"
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Zsh
+        run: |
+          if [ "$RUNNER_OS" == "Linux" ]; then
+            sudo apt-get update
+            sudo apt-get install zsh
+          else
+            brew update
+            brew install zsh
+          fi
+      - uses: actions/setup-haskell@v1.1.4
+        id: setup-haskell-cabal
+        with:
+          ghc-version: ${{ matrix.ghc }}
+          cabal-version: "3.0"
+      - run: cabal update --enable-tests
+      - run: cabal freeze --enable-tests
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ${{ steps.setup-haskell-cabal.outputs.cabal-store }}
+            dist-newstyle
+          key: dependencies-${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles('cabal.project.freeze') }}
+          restore-keys: |
+            dependencies-${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles('cabal.project.freeze') }}
+            dependencies-${{ runner.os }}-${{ matrix.ghc }}-
+      - run: cabal build --enable-tests
+      - run: cabal test --enable-tests
+      - run: cabal haddock --enable-tests | grep "100%" | wc -l | grep -G "[45]" # Fixes issue with different haddock coverage with different ghc versions https://github.com/haskell/haddock/issues/123
+      - run: cabal sdist
+      - if: matrix.ghc == '8.0.2'
+        name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - if: matrix.ghc == '8.0.2'
+        name: Dockerize
+        run: |
+          docker build . -t hapistrano
+          docker run --rm hapistrano --version
```

Comment:
@ibarrae perhaps I'm missing something, but, it looks like we are building the image but not pushing the result to Docker Hub?

Reply:
Yup, that's how the old script was actually. It was never pushing the image so I'm guessing that the purpose was to only check if the image builds properly